src/main/java/org/prebid/server/settings/JdbcApplicationSettings.java

```diff
@@ -77,24 +77,48 @@ public JdbcApplicationSettings(JdbcClient jdbcClient, String selectQuery, String
                 Collections.singletonList(accountId),
                 result -> mapToModelOrError(result, row -> Account.of(row.getString(0), row.getString(1),
                         row.getInteger(2), row.getInteger(3), row.getBoolean(4))),
-                timeout);
+                timeout)
+                .compose(JdbcApplicationSettings::failedIfNull);
     }
 
     /**
      * Runs a process to get AdUnit config by id from database
-     * and returns {@link Future&lt;{@link String}&gt;}
+     * and returns {@link Future&lt;{@link String}&gt;}.
      */
     @Override
     public Future<String> getAdUnitConfigById(String adUnitConfigId, Timeout timeout) {
         return jdbcClient.executeQuery("SELECT config FROM s2sconfig_config where uuid = ? LIMIT 1",
                 Collections.singletonList(adUnitConfigId),
                 result -> mapToModelOrError(result, row -> row.getString(0)),
-                timeout);
+                timeout)
+                .compose(JdbcApplicationSettings::failedIfNull);
+    }
+
+    /**
+     * Transforms the first row of {@link ResultSet} to required object or returns null.
+     * <p>
+     * Note: mapper should never throws exception in case of using
+     * {@link org.prebid.server.vertx.jdbc.CircuitBreakerSecuredJdbcClient}.
+     */
+    private <T> T mapToModelOrError(ResultSet result, Function<JsonArray, T> mapper) {
```

Comment:
The name could have been better for this method as it doesn't produce error anymore